x-pack/plugins/reporting/server/export_types/common/get_conditional_headers.ts

```diff
@@ -7,15 +7,10 @@
 import { ReportingConfig } from '../../';
 import { ConditionalHeaders } from '../../types';
 
-export const getConditionalHeaders = <ScheduledTaskParamsType>({
-  config,
-  job,
-  filteredHeaders,
-}: {
-  config: ReportingConfig;
-  job: ScheduledTaskParamsType;
-  filteredHeaders: Record<string, string>;
-}) => {
+export const getConditionalHeaders = <TaskPayloadType>(
+  filteredHeaders: Record<string, string>,
+  config: ReportingConfig
+) => {
```

Comment:
This seems more type safe with multiple parameters instead of a single options object. It is clearer to see when something isn't used (like `job` in this case)